File: 6주차/징검다리_베베.java
import java.util.Arrays;

public class Solution {
    public int solution(int distance, int[] rocks, int n) {
        int answer = 0;

        Arrays.sort(rocks);

        int left = 1;
        int right = distance;

        while (left <= right) {
            int mid = (left + right) / 2;
            int removedRock = 0;
            int before = 0;

            for (int rock : rocks) {
                if (rock - before < mid) {
                    removedRock++;
                    continue;
                }

                before = rock;
            }

            if (distance - before < mid) {
                removedRock++;
            }

            if (removedRock > n) {
                right = mid - 1;
                continue;
            }

            answer = Math.max(answer, mid);
            left = mid + 1;
        }

        return answer;
    }
}

File: 7,8주차/징검다리_베베.py
def solution(distance, rocks, n):
    rocks.sort()

    left = 1
    right = distance
    answer = 0

    while left <= right:
        mid = (left + right) // 2
        removedRock = 0
        before = 0

        for rock in rocks:
            if rock - before < mid:
                removedRock += 1
                continue

            before = rock

        if distance - before < mid:
            removedRock += 1

        if removedRock > n:
            right = mid - 1
            continue

        answer = max(answer, mid)
        left = mid + 1

    return answer
